### PR TITLE
Unused code cleanup & Discord Rich Presence toggle

### PR DESCRIPTION
**Pull request by Lollyde**
I implemented the toggle in a way where it *does* show up in the ingame mod menu - it doesnt cause any issues (except for it doesnt actually update the the rich presence text for a while), but I'm not sure if it'd be better suited as an option that *doesnt* show up ingame.

### CODE REVIEW

File: Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs [hunk 1]
```diff
@@ -222,6 +222,20 @@ public class CoreModuleSettings : EverestModuleSettings {
...
+        public bool DisableDiscordRichPresence {
+            get => disableDiscordRichPresence;
+            set {
+                if (value) {
+                    Everest.Discord.Disable();
+                } else {
+                    Everest.Discord.Initialize();
+                }
+                disableDiscordRichPresence = value;
+            }
```
Comment (0x0ade) [inline Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs]: There's no protection against this being set to true / false consecutively multiple times. Either implement something in Everest.Discord to take this into account, or (cheaper) `if (value != disableDiscordRichPresence) { ... }`

**Review by 0x0ade — CHANGES_REQUESTED** [review]

File: Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs (hunk 1 above)
Comment (Lollyde) [reply Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs]: added a check to skip the initialization/shutdown if Everest.Discord is already initialized/shut down 👍

**Comment by Lollyde** [comment]
actually, this commit breaks initializing Discord Rich Presence somehow - do not merge yet

File: Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs [hunk 2]
```diff
@@ -222,17 +222,16 @@ public class CoreModuleSettings : EverestModuleSettings {
...
         }
 
-        private bool disableDiscordRichPresence = false;
-
         public bool DisableDiscordRichPresence {
-            get => disableDiscordRichPresence;
+            get => !Everest.Discord.Initialized;
             set {
+                if (value != Everest.Discord.Initialized)
+                    return;
```
Comment (0x0ade) [inline Celeste.Mod.mm/Mod/Core/CoreModuleSettings.cs:229]: This check should ideally be moved to `Everest.Discord` as there *could* be f.e. a mod calling `.Initialize()` without first checking `.Initialized` - it is fine as is though.

**Review by 0x0ade — APPROVED** [review]

**Pull request merged**